apps/interactor/src/state/listeners.ts

```diff
@@ -11,15 +11,36 @@ import { RootState } from './store'
 import textCompletion from '../libs/textCompletion'
 import PromptBuilder from '../libs/memory/PromptBuilder'
 
+let lastTime = new Date().getTime()
+
 const interactionEffect = async (
   dispatch: Dispatch,
   state: RootState,
   servicesEndpoint: string
 ) => {
   try {
+    const response = await fetch(
+      `${servicesEndpoint}/text/strategy/${state.settings.model}`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        cache:
+          new Date().getTime() - lastTime > 600000 ? 'reload' : 'force-cache',
+      }
+    )
+    lastTime = new Date().getTime()
+
+    if (!response.ok || response.status != 200) {
+      throw new Error('Error getting prompt strategy')
+    }
+
+    const data = await response.json()
```

Comment:
You can remove the time tracking and query only once in the page session
Before the function
```typescript
const cacheStrategy = new Map<ModelType, StrategySlug>();
```
And then here:
```suggestion
if (!cacheStrategy.get(state.settings.model)) {
    const response = await fetch(
      `${servicesEndpoint}/text/strategy/${state.settings.model}`,
      {
        method: 'GET',
        headers: {
          'Content-Type': 'application/json',
        },
    )

    if (!response.ok || response.status != 200) {
      throw new Error('Error getting prompt strategy')
    }

    const data = await response.json()
    cacheStrategy.set(state.settings.model, data.strategy as StrategySlug);
}
const strategy = cacheStrategy.get(state.settings.model)
```

Reply:
added in latest commit